test/NosCore.Tests/HandlerTests/ExchangeControllerTests.cs

```diff
@@ -0,0 +1,251 @@
+                    databaseName: Guid.NewGuid().ToString());
+            DataAccessHelper.Instance.InitializeForTest(contextBuilder.Options);
+
+            var items = new List<Item>
+            {
+                new Item { Type = PocketType.Main, VNum = 1012 },
+                new Item { Type = PocketType.Main, VNum = 1013 },
+            };
+
+            _itemBuilderService = new ItemBuilderService(items, new List<IHandler<Item, Tuple<IItemInstance, UseItemPacket>>>());
+            _exchangeService = new ExchangeService(_itemBuilderService, _worldConfiguration);
+
+            _session = new ClientSession(_worldConfiguration,
+                new List<PacketController> { new ExchangePacketController() }, null, _exchangeService);
+
+            _session2 = new ClientSession(_worldConfiguration, new List<PacketController> { new ExchangePacketController() }, null, _exchangeService);
+
+            _session3 = new ClientSession(_worldConfiguration, new List<PacketController> { new ExchangePacketController() }, null, _exchangeService);
+
+            var account1 = new AccountDto { Name = "AccountTest", Password = "test".ToSha512() };
+            var character1 = new CharacterDto
+            {
+                CharacterId = 1,
+                Name = "TestExistingCharacter",
+                Slot = 1,
+                AccountId = account1.AccountId,
+                MapId = 1,
+                State = CharacterState.Active
+            };
+
+            var account2 = new AccountDto { Name = "Account2", Password = "test".ToSha512() };
+            var character2 = new CharacterDto
+            {
+                CharacterId = 2,
+                Name = "TestExistingCharacter2",
+                Slot = 1,
+                AccountId = account2.AccountId,
+                MapId = 1,
+                State = CharacterState.Active
+            };
+
+            var account3 = new AccountDto { Name = "Account3", Password = "test".ToSha512() };
+            var character3 = new CharacterDto
+            {
+                CharacterId = 3,
+                Name = "TestExistingCharacter3",
+                Slot = 1,
+                AccountId = account3.AccountId,
+                MapId = 1,
+                State = CharacterState.Active
+            };
+
+            _session.InitializeAccount(account1);
+            _session2.InitializeAccount(account2);
+            _session3.InitializeAccount(account3);
+            
+            _itemBuilderService = new ItemBuilderService(items, new List<IHandler<Item, Tuple<IItemInstance, UseItemPacket>>>());
+            _handler = new ExchangePacketController(_worldConfiguration, _exchangeService);
+            _handler2 = new ExchangePacketController(_worldConfiguration, _exchangeService);
+            _handler3 = new ExchangePacketController(_worldConfiguration, _exchangeService);
+            _map = new MapInstance(new Map
```

Comment:
don't copy past things you don't need

Reply:
I do need a mapinstance, users should not be able to exchange if they are on different maps